### PR TITLE
Add a custom feature flag model verison saving mechanism

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

As an alternative to #8521 

Adds a mechanism for using Django signals to store model versions in a DB table

stores a feature flag as 

```
"{\"model\": \"posthog.featureflag\", \"pk\": 2, \"fields\": {\"key\": \"6619-query-events-by-date\", \"name\": \"testing\", \"filters\": {\"groups\": [{\"properties\": [], \"rollout_percentage\": null}], \"multivariate\": null}, \"rollout_percentage\": null, \"team\": 1, \"created_by\": 1, \"created_at\": \"2022-02-09T09:30:35.855Z\", \"deleted\": false, \"active\": true}}"
```

## How did you test this code?

hardly at all, just by running it and saving a feature flag over and over

### CODE REVIEW

File: posthog/models/feature_flag.py [hunk 1]
```diff
@@ -101,6 +100,41 @@ def get_filters(self):
...
+    team_id = models.PositiveIntegerField(null=False)
+    created_by_name = models.TextField(null=False)
+    created_by_id = models.PositiveIntegerField(null=False)
+
+
+@receiver(post_save, sender=FeatureFlag, dispatch_uid="save_feature_flag_version")
+def save_version(sender, instance: FeatureFlag, **kwargs):
+    try:
+        version = ModelVersion(
+            #  ¯\_(ツ)_/¯ serialize the instance as a list and then chop off the square braces
```
Comment (mariusandra) [inline posthog/models/feature_flag.py:124]: 👀

File: posthog/migrations/0206_add_revision_model.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    dependencies = [
+        ("posthog", "0205_auto_20220204_1748"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="ModelVersion",
+            fields=[
+                ("id", models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name="ID")),
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/migrations/0206_add_revision_model.py:17]: yep

File: posthog/models/feature_flag.py (hunk 1 above)
Comment (pauldambra) [reply posthog/models/feature_flag.py:124]: Took 60% of the time to write the example working through all the ways people claim you can serialize a model 🐰

File: posthog/models/feature_flag.py [hunk 3]
```diff
@@ -101,6 +100,41 @@ def get_filters(self):
...
+
+    model_state = models.JSONField(null=False)
+    model_name = models.fields.TextField(null=False)
+    # to avoid an integer version field for ordering revisions
+    versioned_at: models.DateTimeField = models.DateTimeField(default=timezone.now)
+
+    # don't store a reference cos the team or user might get deleted.  ¯\_(ツ)_/¯
+    # audit log should be immutable
+    team_id = models.PositiveIntegerField(null=False)
+    created_by_name = models.TextField(null=False)
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/models/feature_flag.py:116]: Ha, great minds think alike https://github.com/PostHog/posthog/issues/8545#issuecomment-1035511330

**Pull request closed without merging**